File: 第三章_字符串/字符串(3)/合法括号序列判断练习题/Parenthesis.cpp
class Parenthesis {
public:
    bool chkParenthesis(string A, int n) {
        int i;
        int num=0;
        for(i=0;i<n;i++)
        {
            if(A[i]=='(')
            {
                num++;
            }
            if(A[i]==')')
            {
                num--;
            }
            if(num<0)
            {
                return false;
            }
        }
 
        return num==0;
    }
};

File: 第三章_字符串/字符串(3)/合法括号序列判断练习题/Parenthesis.java
import java.util.*;

public class Parenthesis {
    public boolean chkParenthesis(String A, int n) {
        int num=0;
    	
    	for(int i=0;i<n;i++){
        	if (A.charAt(i)=='(') {
				num++;
			}
        	if (A.charAt(i)==')') {
				num--;
			}
        	if (num<0) {
				return false;
			}
        }
    	return num==0;
    }
}